### PR TITLE
feature(frontend): Required Gates

### PR DESCRIPTION
**Pull request by xoscar**
This PR includes changes for the FE and backend to support the required gates feature

## Changes

- Adds the Test runner and Required Gates Result models
- Encapsulates the status icon into a single component
- Updates backend to support error messaging for failure gates from any test
- Displays tooltips for test run results
- Includes Test Run settings page and logic

## Fixes

- #2777 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [] updated the docs
- [ ] added a test

https://www.loom.com/share/2ef5aab0ed5d4ac6ac9c729efafab53d

### CODE REVIEW

File: web/src/components/RunStatusIcon/RunStatusIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import TestRun, {isRunStateFailed, isRunStateFinished, isRunStateStopped} from 'models/TestRun.model';
+import RequiredGatesResult from 'models/RequiredGatesResult.model';
+import * as S from './RunStatusIcon.styled';
+import RunTooltipTitle from './RunTooltipTitle';
+
+interface IProps {
+  state: TestRun['state'];
+  requiredGatesResult: RequiredGatesResult;
+}
+
```
Comment (xoscar) [inline web/src/components/RunStatusIcon/RunStatusIcon.tsx:11]: This component encapsulates the run status icon that should be used everywhere in the app

File: web/src/components/Settings/TestRunner/TestRunnerForm.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import {useCallback, useEffect} from 'react';
```
Comment (xoscar) [inline web/src/components/Settings/TestRunner/TestRunnerForm.tsx:1]: Adding the settings form for the test runner options

File: web/src/models/RequiredGatesResult.model.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const RequiredGatesResult = ({
+  required = [],
+  failed = [],
+  passed = true,
+}: TRawRequiredGatesResult): RequiredGatesResult => ({
+  required,
+  failed,
+  passed,
+  requiredFailedGates: required.filter(gate => failed.includes(gate)),
```
Comment (xoscar) [inline web/src/models/RequiredGatesResult.model.ts:19]: Calculates the failed gates that are required from the model so we can use them elsewhere

File: web/src/redux/apis/endpoints/Setting.endpoint.ts [hunk 4]
```diff
@@ -63,6 +64,15 @@ const ConfigEndpoint = (builder: TTestApiEndpointBuilder) => ({
     providesTags: () => [{type: TracetestApiTags.SETTING, id: ResourceType.AnalyzerType}],
     transformResponse: (rawLinter: TRawLinter) => Linter(rawLinter),
   }),
+  getTestRunner: builder.query<TestRunner, unknown>({
```
Comment (xoscar) [inline web/src/redux/apis/endpoints/Setting.endpoint.ts:67]: I was thinking we can simplify the endpoints for settings and the overall resources, we can think about a clever way to handle that in the future

File: web/src/components/Settings/TestRunner/RequiredGatesInput.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  return (
+    <div>
+      <Typography.Title level={3}>Required Gates</Typography.Title>
+      <S.SwitchListContainer>
+        {supportedGates.map(gate => (
+          <S.SwitchContainer>
+            <Switch checked={value.includes(gate)} onChange={isChecked => handleChange(gate, isChecked)} />
+            <Typography.Text>
+              {ToTitle(gate)} <TooltipQuestion margin={6} title={SupportedRequiredGatesDescription[gate]} />
+            </Typography.Text>
```
Comment (jorgeepc) [inline web/src/components/Settings/TestRunner/RequiredGatesInput.tsx:34]: We could use a `label` here

File: web/src/constants/Common.constants.ts [hunk 6]
```diff
@@ -22,6 +22,7 @@ export const TRACE_DOCUMENTATION_URL =
 export const ADD_TEST_URL = 'https://docs.tracetest.io/web-ui/creating-tests';
 export const ADD_TEST_OUTPUTS_DOCUMENTATION_URL = 'https://docs.tracetest.io/web-ui/creating-test-outputs';
 export const ANALYZER_DOCUMENTATION_URL = 'https://docs.tracetest.io/analyzer/concepts';
+export const TEST_RUNNER_DOCUMENTATION_URL = 'https://docs.tracetest.io/analyzer/concepts';
```
Comment (jorgeepc) [inline web/src/constants/Common.constants.ts:25]: Should we change this now?

File: web/src/models/TestRunner.model.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+export enum SupportedRequiredGates {
+  AnalyzerScore = 'analyzer-score',
+  AnalyzerRules = 'analyzer-rules',
+  TestSpecs = 'test-specs',
+}
+
+export const SupportedRequiredGatesDescription = {
+  [SupportedRequiredGates.AnalyzerScore]: 'Test Runs will be marked as failed if the Analyzer Score is below the configured threshold',
+  [SupportedRequiredGates.AnalyzerRules]: 'Test Runs will be marked as failed if the Error Level Analyzer Rules are not met',
+  [SupportedRequiredGates.TestSpecs]: 'Test Runs will be marked as failed if on of the defined Test Specs fail',
```
Comment (jorgeepc) [inline web/src/models/TestRunner.model.ts:17]: Test Runs will be marked as failed if `one` of ?

File: web/src/redux/apis/endpoints/Setting.endpoint.ts (hunk 4 above)
Comment (jorgeepc) [reply web/src/redux/apis/endpoints/Setting.endpoint.ts:67]: Like that!

**Review by jorgeepc — APPROVED** [review]
Great work! Added a couple of comments

**Pull request merged**